### PR TITLE
Updated front end

### PR DESCRIPTION
**Pull request by nesh325**

### CODE REVIEW

File: Client/src/App.js [hunk 1]
```diff
@@ -1,12 +1,31 @@
...
+import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
 
-function App() {
+const App = () => {
   return (
-    <div>
-      <h2>Welcome to React App</h2>
-      <h3>Date : {new Date().toDateString()}</h3>
+    <div className="bg-gray-200 w-full min-h-screen">
+      <Router>
```
Comment (Sumit-Kap) [inline Client/src/App.js:11]: Here we could create an array of routes to make it scalable where on addition of routes only entry in that array should be updated else would be kept intact something like this 

const routes = [
	{ path: '/', component: HomeView, renderOnServer: true},
	{ path: '/profile', component: LoginView, renderOnServer: true},
	{ path: '/saved', component: ChatView }
]

File: Client/src/App.js [hunk 2]
```diff
@@ -1,12 +1,31 @@
...
+        <Switch>
+          <Route exact path="/">
+            <Home />
+          </Route>
+          <Route path="/profile">
```
Comment (Sumit-Kap) [inline Client/src/App.js:16]: you can also pass component prop inside your <Route/> declaration see below example. Read this https://reactrouter.com/web/api/Route
```suggestion
          <Route path="/profile" component={Profile}>
```

File: Client/src/Components/NavBar.jsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import React, { useState } from "react";
+
+import { MdArrowDropDown } from "react-icons/md";
+import { SiTwitter } from "react-icons/si";
+import { useHistory } from "react-router-dom";
+
+const NavBar = () => {
+  let history = useHistory("/");
+  let [active, setActive] = useState(1);
```
Comment (Sumit-Kap) [inline Client/src/Components/NavBar.jsx:9]: can't we use an enum instead of numbers here?

File: Client/src/Components/NavBar.jsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+              : "text-gray-500 text-md font-normal"
+          }`}
+          onClick={(e) => handleClick(e, "saved", 3)}
+        >
+          Saved
+        </button>
+      </div>
+      <div className="flex-row items-center p-4 sm:px-12">
+        <div className="bg-blue-600 w-8 h-8 rounded-lg"></div>
+        <p className="font-roboto font-medium	text-sm ml-2">John Doe</p>
```
Comment (Sumit-Kap) [inline Client/src/Components/NavBar.jsx:57]: power this name through constant file.

File: Client/src/Components/NewTweet.jsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import { MdImage, MdPublic } from "react-icons/md";
+
+const NewTweet = () => {
+  return (
+    <div className="h-auto p-5 bg-white rounded-md ">
+      <p className="font-poppins text-sm font-medium">Tweet something</p>
```
Comment (Sumit-Kap) [inline Client/src/Components/NewTweet.jsx:7]: power all the string values through constant file.

File: Client/src/index.js [hunk 6]
```diff
@@ -1,5 +1,7 @@
+require("file-loader?name=[name].[ext]!../build/index.html");
 import React from "react";
-import ReactDom from "react-dom";
+import ReactDOM from "react-dom";
 import App from "./App";
+import "./styles.css";
 
-ReactDom.render(<App />, document.getElementById("app"));
+ReactDOM.render(<App />, document.getElementById("app"));
```
Comment (Sumit-Kap) [inline Client/src/index.js:7]: This got changed in react 18, check this link https://javascript.works-hub.com/learn/react-18-is-here-whats-new-9b46a

File: node_modules/react-icons/LICENSE [hunk 7]
```diff
@@ -0,0 +1,74 @@
+Copyright 2018 kamijin_fanta <kamijin@live.jp>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
```
Comment (Sumit-Kap) [inline node_modules/react-icons/LICENSE:5]: why are you checking your node_modules files here?

**Review by Sumit-Kap — CHANGES_REQUESTED** [review]
Please remove all your node_modules files from here. add node_modules in your .gitignore.

File: Client/src/App.js (hunk 1 above)
Comment (Sumit-Kap) [reply Client/src/App.js:11]: Now use this to iterate over the routes for example.

<Switch>
				{
					routes.map((route, i)=>{
						return <Route key={i} exact path ={route.path} component={route.component} />
					})
				}
			</Switch>

Comment (nesh325) [reply Client/src/App.js:11]: Thank you, I like this approach a lot. Fixed and will update on next push

File: Client/src/Components/NavBar.jsx (hunk 3 above)
Comment (nesh325) [reply Client/src/Components/NavBar.jsx:9]: How would I do enums? Something like 
Something like - 

const NavLinkTitles = { Home:1, ...}

File: Client/src/Components/NavBar.jsx (hunk 4 above)
Comment (nesh325) [reply Client/src/Components/NavBar.jsx:57]: Do you mean to put "John Doe" in const variable?

File: Client/src/index.js (hunk 6 above)
Comment (nesh325) [reply Client/src/index.js:7]: `const container = document.getElementById("app");

const root = ReactDOM.createRoot(container, { hydrate: true });

root.render(<App />);`
Doesn't seem to work. Updated react to 18 as well

File: node_modules/react-icons/LICENSE (hunk 7 above)
Comment (nesh325) [reply node_modules/react-icons/LICENSE:5]: Removed

File: Client/src/Components/NavBar.jsx (hunk 4 above)
Comment (nesh325) [reply Client/src/Components/NavBar.jsx:57]: I'm not sure about this. Make a file with const text and then import and insert one by one?

**Pull request merged**